contracts/VoucherHub.sol

```diff
@@ -177,6 +177,25 @@ contract VoucherHub is
         emit VoucherCreated(voucherAddress, owner, voucherExpiration, voucherType, value);
     }
 
+    /**
+     * Top up a voucher by increasing its balance and extending the validity
+     * period before expiration.
+     * @param voucher The address of the voucher.
+     * @param value The amount of credits to top up.
+     */
+    function topUpVoucher(address voucher, uint256 value) external onlyRole(VOUCHER_MANAGER_ROLE) {
+        VoucherHubStorage storage $ = _getVoucherHubStorage();
+        require($._isVoucher[voucher], "VoucherHub: unknown voucher");
+        if (value > 0) {
+            _mint(voucher, value); // VCHR
+            IERC20($._iexecPoco).transfer(voucher, value); // SRLC
+        }
+        uint256 voucherExpiration = block.timestamp +
+            $.voucherTypes[Voucher(voucher).getType()].duration;
+        Voucher(voucher).setExpiration(voucherExpiration);
```

Comment:
Is it cheaper to use one instance of `Voucher(voucher)`?

Reply:
Exactly same price (min, max & avg)!
Do you want a syntax update here?

Reply:
No it's ok like that. Thank you testing it.